File: scripting/engine/Playground/ecs/components/SunComponent.swift
@component
public struct SunComponent: Codable {
    public var direction: Vector3
    public var colour: Rendering.Colour
    public var intensity: Float

    public init(
        direction: Vector3,
        colour: Rendering.Colour,
        intensity: Float
    ) {
        self.direction = direction
        self.colour = colour
        self.intensity = intensity
    }
}

File: engine/managed/PlaygroundAssembly/PlaygroundAssembly/ECS/Rendering/SunComponent.cs
using Playground.Core.Ecs;
using PlaygroundAssembly.ECS.Core;
using PlaygroundAssembly.Math;
using PlaygroundAssembly.Rendering;

namespace PlaygroundAssembly.ECS.Rendering;

[EcsComponent]
public struct SunComponent
{
    public Vector3 Direction;
    public Colour Colour;
    public float Intensity;
}
